src/yarp.c

```diff
@@ -123,6 +123,14 @@ char_is_whitespace(const char *c) {
   return char_is_non_newline_whitespace(c) || *c == '\n';
 }
 
+static inline bool
+char_is_symbol(const char *c) {
```

Comment:
This could also be done with `ispunct` but that would involve pulling in `ctype.h`. Since there was already other functions that did char checking I felt this was the best path forward but let me know if pulling in `ctype.h` is preferred.